packages/compass-editor/src/autocompleter.ts

```diff
@@ -1,178 +1,25 @@
- * properties that completer is using
- * @internal
- */
-export type Completion = {
-  value: string;
-  version: string;
-  meta: Meta;
-  description?: string;
-  snippet?: string;
-  score?: number;
-};
-
-function matchesMeta(filter: string[], meta: string) {
-  const metaParts = meta.split(':');
-  return filter.some((metaFilter) => {
-    const filterParts = metaFilter.split(':');
-    return (
-      filterParts.length === metaParts.length &&
-      filterParts.every((part, index) => {
-        return part === '*' || part === metaParts[index];
-      })
-    );
-  });
-}
-
-export function createCompletionFilter(
-  prefix: string,
-  serverVersion: string,
-  filterMeta?: string[]
-) {
-  const currentServerVersion =
-    serverVersion.match(/^(?<version>\d+?\.\d+?\.\d+?)/)?.groups?.version ??
-    DEFAULT_SERVER_VERSION;
-  return ({ value, version: minServerVersion, meta }: Completion) => {
-    return (
-      value.toLowerCase().startsWith(prefix.toLowerCase()) &&
-      gte(currentServerVersion, minServerVersion) &&
-      (!filterMeta || matchesMeta(filterMeta, meta))
-    );
-  };
-}
-
-export type CompleteOptions = {
-  // Current server version (default is 999.999.999)
-  serverVersion?: string;
-  // Additional fields that are part of the document schema to add to
-  // autocomplete as identifiers and identifier references
-  fields?: (string | { name: string; description?: string })[];
-  // Filter completions by completion value type
-  meta?: (Meta | 'field:*' | 'accumulator:*' | 'expr:*')[];
-};
-
-export type CompletionResult = {
-  // Autocomplete value that prefix was matched on
-  value: string;
-  // Value type
-  meta?: string;
-  // Longer description
-  description?: string;
-  // String representing a possible snippet completion
-  snippet?: string;
-  // For ace compat
-  score: number;
-};
-
-function isValidIdentifier(identifier: string) {
-  // Quick check for common case first
-  if (/[.\s"'()[\];={}]/.test(identifier)) {
-    return false;
-  }
-  try {
-    // Everything else we check using eval as regex methods of checking are quite
-    // hard to do (see https://mathiasbynens.be/notes/javascript-identifiers-es6)
-    // eslint-disable-next-line @typescript-eslint/no-implied-eval
-    new Function(`"use strict";let ${identifier};`);
-    return true;
-  } catch {
-    return false;
-  }
-}
-
-/**
- * Helper method to conditionally wrap completion value if it's not a valid
- * identifier
- */
-export function wrapField(field: string, force = false) {
-  return force || !isValidIdentifier(field)
-    ? `"${field.replace(/["\\]/g, '\\$&')}"`
-    : field;
-}
-
-function normalizeField(
-  field: string | { name: string; description?: string }
-) {
-  return typeof field === 'string'
-    ? { value: field }
-    : {
-        value: field.name,
-        description: field.description,
-      };
-}
+// TODO: add type exports from the package
```

Comment:
Do we want to link a ticket for this? Fine as is too.
```suggestion
// TODO: use type exports from the @mongodb-js/mongodb-constants package.
```

Reply:
I'll do it in the same PR, just waiting for CI in devtools-shared